src/domain/eventSearch/utils.tsx

```diff
@@ -263,30 +311,41 @@ export const getEventSearchVariables = ({
     keywordAnd.push('yso:p4354');
   }
 
-  const keywordsParamName = MAPPED_KEYWORD_TERMS[searchSource];
-  const mappedCategories: string[] = categories
-    .map((category) => MAPPED_CATEGORIES[searchSource][category])
-    .filter((e) => e);
-  const mappedHobbyTypes: string[] =
-    hobbyTypes
-      ?.map((hobbyType) => MAPPED_COURSE_HOBBY_TYPES[hobbyType])
-      .filter((e) => e) || [];
+  const categoriesParamName = MAPPED_KEYWORD_TERMS[searchSource];
+  let mappedCategories: string[] = [];
+  categories.map(
```

Comment:
In this scenario .forEach would fit better because we are not using the return value of map, and the point of map is to use the return value.

But I think reduce would fit this use case the best.

```
  const mappedCategories = categories.reduce<string[]>(
    (prev, category) =>
      prev.concat(MAPPED_CATEGORIES[searchSource][category]?.split(',')),
    []
  );
```

or even

```
  const mappedCategories = categories.reduce<string[]>(
    (prev, category) => [
      ...prev,
      ...MAPPED_CATEGORIES[searchSource][category]?.split(','),
    ],
    []
  );
```

Reply:
might be even better if we get the map outside of reduce:

```
  const categoryMap = MAPPED_CATEGORIES[searchSource];
  const mappedCategories = categories.reduce<string[]>(
    (prev, category) => prev.concat(categoryMap[category]?.split(',')),
    []
  );
```

Reply:
i will check